libraries/hotstuff/qc_chain.cpp

```diff
@@ -128,32 +128,22 @@ namespace eosio { namespace hotstuff {
 #endif
    }
 
-   uint32_t qc_chain::positive_bits_count(fc::unsigned_int value){
-      boost::dynamic_bitset b(21, value);
-      uint32_t count = 0;
-      for (boost::dynamic_bitset<>::size_type i = 0; i < b.size(); i++){
-         if (b[i]==true)count++;
-      }
-      return count;
+   uint32_t qc_chain::positive_bits_count(const boost::dynamic_bitset<>& finalizers) {
+      return finalizers.count(); // the number of bits in this bitset that are set.
    }
 
-   fc::unsigned_int qc_chain::update_bitset(fc::unsigned_int value, name finalizer ) {
-
-      fc_tlog(_logger, " === update bitset ${value} ${finalizer}",
-           ("value", value)
-           ("finalizer", finalizer));
+   boost::dynamic_bitset<> qc_chain::update_bitset(const boost::dynamic_bitset<>& finalizer_set, name finalizer ) {
 
-      boost::dynamic_bitset b( 21, value );
+      boost::dynamic_bitset b( finalizer_set );
       vector<name> finalizers = _pacemaker->get_finalizers();
       for (size_t i = 0; i < finalizers.size();i++) {
          if (finalizers[i] == finalizer) {
             b.flip(i);
```

Comment:
should it be `b.set(i)`?

Reply:
One would assume so. @systemzax ?

Reply:
Yes, .set(i). Not sure why I had flip there, doesn't make sense.